src/ifcconvert/IfcConvert.cpp

```diff
@@ -373,30 +455,63 @@ int main(int argc, char** argv) {
 	// true return value guarantees that a successfully processed product is 
 	// available. 
 	do {
-		const IfcGeom::Element<double>* geom_object = context_iterator.get();
-		
-		if (serializer->isTesselated()) {
-			serializer->write(static_cast<const IfcGeom::TriangulationElement<double>*>(geom_object));
-		} else {
-			serializer->write(static_cast<const IfcGeom::BRepElement<double>*>(geom_object));
-		}
-		
-		const int progress = context_iterator.progress() / 2;
-		if (old_progress!= progress) Logger::ProgressBar(progress);
-		old_progress = progress;
-	
-	} while (context_iterator.next());
+        IfcGeom::Element<real_t> *geom_object = context_iterator.get(true); // true == take ownership, we will clean up ourselves
+        geometries.push_back(geom_object);
+        const int progress = context_iterator.progress() / 2;
+        if (old_progress != progress) Logger::ProgressBar(progress);
+        old_progress = progress;
+
+        if (center_model) {
```

Comment:
Of course this is a valuable feature. But considering it is only based on placement/insertionpoint and not on actual geometry bounding box. I think it is better to precompute this based on the product object placements prior to doing the geometry conversion loop. Converting them into a `gp_Trsf` beforehand is trivial, shouldn't take much time and the result is cached. This way we do not need to keep elements in memory and the streaming serializers such as OBJ can start right away. If you prefer I can do this after the merge though.